addons/mrp/migrations/10.0.2.0/pre-migration.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent <http://www.eficent.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+from odoo import tools
+
+_column_copies = {
+    'mrp_production': [
+        ('state', None, None)
+    ],
+    'mrp_workorder': [
+        ('sequence', None, None)
+    ],
+}
+
+_column_renames = {
+    'stock_move': [
+        ('consumed_for', None),
+    ],
+}
+
+_field_renames = [
+    ('mrp.bom', 'mrp_bom', 'product_uom', 'product_uom_id'),
+    ('mrp.bom.line', 'mrp_bom_line', 'product_uom', 'product_uom_id'),
+    ('mrp.production', 'mrp_production', 'date_planned', 'date_planned_start'),
+    ('mrp.production', 'mrp_production', 'move_lines', 'move_raw_ids'),
+    ('mrp.production', 'mrp_production', 'move_created_ids2', 'move_finished_ids'),
+    ('mrp.production', 'mrp_production', 'product_uom', 'product_uom_id'),
+    ('mrp.production', 'mrp_production', 'workcenter_lines', 'workorder_ids'),
+    ('mrp.routing', 'mrp_routing', 'workcenter_lines', 'operation_ids'),
+    ('mrp.workcenter', 'mrp_workcenter', 'capacity_per_cycle', 'capacity'),
+]
+
+_field_renames2 = [
+    ('mrp.workorder', 'mrp_workorder', 'date_planned', 'date_planned_start'),
+]
+
+_model_renames = [
+    ('mrp_production_workcenter_line', 'mrp_workorder'),
+]
+
+_xmlid_renames = [
+    ('base.menu_mrp_config', 'mrp.menu_mrp_config'),
+    ('base.menu_mrp_root', 'mrp.menu_mrp_root'),
+]
+
+
+def prepopulate_fields(cr):
+    cr.execute(
+        """
+        ALTER TABLE mrp_production
+        ADD COLUMN picking_type_id INTEGER
+        """
+    )
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    cr = env.cr
+    tools.drop_view_if_exists(cr, 'mrp_workorder')
+    openupgrade.rename_models(cr, _model_renames)
```

Comment:
This is renaming models based on data that's actually about tables, not models.
`_model_renames` should be renamed to `_table_renames` and `rename_tables` called with it, not `rename_models`. Additionally might have to do something about the model information as well, though it's already filled by the previous VIEW backed model, maybe that can be deleted and then the model rename done too.

Without these changes, the migration actually fails for me, because there will be no mrp_workorder table at all, so first the sequence deletion from _column_copies fails, and afterwards post-migration on mrp_workorder as well.

Reply:
[FIX] without model_renames seems to already fix this up, but then "[ADD] More migration scripts" pushed later messes it up again